### PR TITLE
Implements the command to generate storage invoices

### PR DESCRIPTION
**Pull request by knikolla**
This PR implements the management command `coldfront calculate_quota_unit_hours --start <start> --end <end>` that generates the  CSV invoice for the given period.

Some minor polish is still missing but it's operational.

### CODE REVIEW

File: src/coldfront_plugin_cloud/management/commands/calculate_storage_gb_hours.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                msg = f'Starting billing for for allocation {allocation_str}.'
+                logger.debug(msg)
+
+                for attr, price_per_unit in [
+                    (attributes.QUOTA_LIMITS_EPHEMERAL_STORAGE_GB, 1)
+                ]:
+                    time = utils.calculate_quota_unit_hours(allocation, attr, start, end)
+                    billed = time * price_per_unit
+                    if billed > 0:
+                        csv_invoice_writer.writerow(
```
[the remark this answers is not in the document]
Comment (knikolla) [reply src/coldfront_plugin_cloud/management/commands/calculate_storage_gb_hours.py]: I refactored this part to use a `dataclass` and wrote some simple logic to get a list of headers and values.

File: src/coldfront_plugin_cloud/management/commands/calculate_storage_gb_hours.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        openshift_resources = Resource.objects.filter(
+            resource_type=ResourceType.objects.get(
+                name='OpenShift'
+            )
+        )
+        openshift_allocations = Allocation.objects.filter(
+            resources__in=openshift_resources
+        )
+
+        with open('invoices.csv', 'w', newline='') as f:
```
[the remark this answers is not in the document]
Comment (knikolla) [reply src/coldfront_plugin_cloud/management/commands/calculate_storage_gb_hours.py]: Done

File: src/coldfront_plugin_cloud/utils.py [hunk 3]
```diff
@@ -43,3 +52,99 @@ def get_sanitized_project_name(project_name):
...
+            print(f"Last event at {last_event_time}, cr created at {created}, approved at {event_time}")
+        else:
+            seconds_since_last_event = math.ceil((event_time - last_event_time).total_seconds())
+            value_times_seconds += seconds_since_last_event * last_event_value
+
+        last_event_time = event_time
+        last_event_value = int(event.value)
+
+    # No special logic is required after processing the last event, just
+    # continue treating the
```
[the remark this answers is not in the document]
Comment (knikolla) [reply src/coldfront_plugin_cloud/utils.py]: Rewrote the comment.

File: src/coldfront_plugin_cloud/utils.py [hunk 4]
```diff
@@ -43,3 +52,99 @@ def get_sanitized_project_name(project_name):
...
+
+def calculate_quota_unit_hours(allocation, attribute, start, end):
+    """Returns unit*hours of quota allocated in a given period.
+
+    Calculation is rounded up by the hour and tracks the history of change
+    requests.
+
+    :param attribute: Name of the attribute to calculate.
+    :param start: Start time to being calculation.
+    :param end: Optional. End time for calculation.
```
[the remark this answers is not in the document]
Comment (knikolla) [reply src/coldfront_plugin_cloud/utils.py]: Done.

**Comment by joachimweyl** [comment]
@naved001 & @larsks it looks like all of the Change Requests have been resolved. Are there any other changes you wish to make before approving this PR?

**Review by larsks — APPROVED** [review]

**Pull request merged**